### PR TITLE
Added template for pricing page

### LINKED ISSUE #291: pricing page
A site containing additional details to the available membership models.

### PR DESCRIPTION
**Pull request by ThorbenStaufer**
Clicking on the "learn more" link in the pricing section of the landing page will now redirect the user to the pricing page.
This page contains all neccessary placeholders to inform the user about our pricing strategy and lets them also apply via this page.

The translation function in the PricingObject component currently displays an error, even though it works perfectly fine.
@Bushuo do you have an idea how we could fix this?

Resolves #291.

## Basics

- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

**Comment by Bushuo** [comment]
@ThorbenStaufer 
> The translation function in the PricingObject component currently displays an error, even though it works perfectly fine.
@Bushuo do you have an idea how we could fix this?

I don't see an error locally. Could you please send me your console output?

**Comment by Bushuo** [comment]
```ts
type OptLtr = 'A' | 'B';
type OptNr = 'one' | 'two';

interface PricingOptionProps {
  title: `pricing:Option${OptLtr}.title`;
  info: `pricing:Option${OptLtr}.info`;
  advantages: readonly [
    `pricing:Option${OptLtr}.advantages.${OptNr}`,
    `pricing:Option${OptLtr}.advantages.${OptNr}`,
  ];
  isSelected: boolean;
  onClickHandler(id: string): void;
}
```

```ts
const initialOptions = [
    {
      title: 'pricing:OptionA.title' as const,
      info: 'pricing:OptionA.info' as const,
      advantages: ['pricing:OptionA.advantages.one', 'pricing:OptionA.advantages.two'] as const,
      isSelected: false,
    },
    {
      title: 'pricing:OptionB.title' as const,
      info: 'pricing:OptionB.info' as const,
      advantages: ['pricing:OptionB.advantages.one', 'pricing:OptionB.advantages.two'] as const,
      isSelected: false,
    },
  ];
```

should fix the error.
I can look at it again, later today after sleep xD

**Comment by markus2330** [comment]
jenkins build please

**Comment by Bushuo** [comment]
Hi, please take a look at #296 and merge it into this PR branch.

File: frontend/src/features/pricing_page/components/PricingOption.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { DefaultLanguage } from '@/config/i18n';
+
+interface PricingOptionProps {
+  /** the translated strings of the pricing option. */
+  pricingOption: DefaultLanguage['pricing']['options'][number];
+  /** a flag indicating whether this option was selected. */
+  isSelected: boolean;
+  /** the index to determine the selection. */
+  index: number;
```
Comment (Bushuo) [inline frontend/src/features/pricing_page/components/PricingOption.tsx:9]:
```suggestion
```
Sorry, I added this by mistake. We do not need the index inside the component.

File: frontend/src/features/pricing_page/routes/PricingPage.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const pricingOptions = t('pricing:options', { returnObjects: true });
+  const [selectedOption, setSelectedOption] = useState(NaN);
+
+  const options = pricingOptions.map((option, i) => {
+    return (
+      <PricingOption
+        key={option.title}
+        pricingOption={option}
+        isSelected={selectedOption === i}
+        index={i}
```
Comment (Bushuo) [inline frontend/src/features/pricing_page/routes/PricingPage.tsx:21]:
```suggestion
```
Same here.

**Review by Bushuo — APPROVED** [review]
Everything looks great 👍

**Comment by markus2330** [comment]
There are unfortunately many conflicts and the template was not in a state that we could use it for the release. So let us close this for now, maybe we can reuse the component later. It looks nice.

**Pull request closed without merging**